plutus-core/exe/Main.hs

```diff
@@ -248,65 +295,30 @@ getCborInput (FileInput file) = BSL.readFile file
 
 
 -- Load a PLC AST from a CBOR file
-loadPlcFromCborFile :: Input -> IO PlainProgram
-loadPlcFromCborFile inp = do
-  p <- getCborInput inp -- The type is constrained in the Right case below.
-  case deserialiseOrFail p of
-    Left (DeserialiseFailure offset msg) ->
-        do
-          putStrLn $ "Deserialisation failure at offset " ++ show offset ++ ": " ++ msg
-          exitFailure
-    Right (r::PlainProgram) -> return r
+loadPlcFromCborFile :: Typing -> Input -> IO (Program ())
+loadPlcFromCborFile typing inp =
+    case typing of
+         Typed -> getCborInput inp <&> deserialiseOrFail >>= \case
+                Left e                     -> failWith e
+                Right (r::TypedProgram ()) -> return (TypedProgram r)
+              -- ^ The type constraint on r is required so that deserialiseOrFail knows what type it should be deserialising to.
+         Untyped -> getCborInput inp <&> deserialiseOrFail >>= \case
+                Left e                       -> failWith e
+                Right (r::UntypedProgram ()) -> return (UntypedProgram r)
```

Comment:
It seems like we should be able to easily abstract that into a function taking a `program ann -> Program ann` argument to wrap the result with. Not too important, though.

Reply:
Hmm.  That's actually better. I thought we'd need some horrible type annotations to tell `deserialiseOrFail` which type it should be deserialising to, but in fact it magically knows that and you can get rid of the type annotations completely.